hypervisor/dm/vpci/vmcs9900.c

```diff
@@ -134,14 +134,14 @@ static void init_vmcs9900(struct pci_vdev *vdev)
 	mmio_vbar->size = 0x1000U;
 	mmio_vbar->base_gpa = pci_cfg->vbar_base[MCS9900_MMIO_BAR];
 	mmio_vbar->mask = (uint32_t) (~(mmio_vbar->size - 1UL));
-	mmio_vbar->fixed = (uint32_t) (mmio_vbar->base_gpa & PCI_BASE_ADDRESS_MEM_MASK);
+	mmio_vbar->fixed = 0U;
```

Comment:
It is named as fixed, so we should program at initial time, and never change later on.

Reply:
vbar value at last is (address & vbar.mask) | vbar.fixed. 
PCI-Vuart's vbar is 32-bit mmio type, suppose unprefetchable, that means vbar[bit 3:0] must be 0b0000,  
And  suppose allow guest OS re-program vbar with any address in vbar[31:4], then pci-vuart vbar don't have any fixed value.

vbar.fixed =0U is correct for 32-bit mmio bar type , former value 'base_gpa & PCI_BASE_ADDRESS_MEM_MASK' is wrong

Reply:
So you mean this is not "fixed" but "bar_type"?   And 0U is actually PCIM_BAR_MEM_32?
Then we should largly reshuffle the current code.

Reply:
About vbar.mask & vbar.fixed, they are more than 'type'. I think if the vbar bit should always be '1' for guest, then set the same bit in vbar.fixed, if the bit should always be '0', then chear the same bit in vbar.mask. 

The correct mask & fixed value should support:
1. indetify the bar type, (bit3:0) 
2. mmio address range, for example if only allow 1G~2G, should set  bit31 in 'fixed' to avoid using 0~2G address. And  clear bit31 in 'mask' to avoid using 2G~4G
3. Sizing vbar: when guest write '~0' and read should return '~(bar_size - 1)'

So for 32-bit mmio type, un-prefetchable, allow re-program to 0~4G. vbar.mask = ~(size -1), vbar.fixed=0, can satisfy the 3 requirement.